esda/correlogram.py

```diff
@@ -0,0 +1,150 @@
+# Spatial Correlograms
+
+import geopandas as gpd
+import pandas as pd
+import numpy as np
+from joblib import Parallel, delayed
+from libpysal.cg.kdtree import KDTree
+from libpysal.weights import KNN, DistanceBand
+from libpysal.weights.util import get_points_array
+
+from .geary import Geary
+from .getisord import G
+from .moran import Moran
+
+
+def _get_autocorrelation_stat(inputs):
+    """helper function for computing parallel autocorrelation statistics
+
+    Parameters
+    ----------
+    inputs : tuple
+        tuple of (y, tree, W, statistic, STATISTIC, dist, weights_kwargs, stat_kwargs)
+
+    Returns
+    -------
+    pandas.Series
+        a pandas series with the computed autocorrelation statistic and its simulated p-value
+    """
+    (
+        y,  # y variable
+        tree,  # kd tree
+        W,  # weights class DistanceBand or KNN
+        STATISTIC,  # class of statistic (Moran, Geary, etc)
+        dist,  # threshold/k parameter for the weights
+        weights_kwargs,  # additional args
+        stat_kwargs,  # additional args
+    ) = inputs
+
+    w = W(tree, dist, silence_warnings=True, **weights_kwargs)
+    autocorr = STATISTIC(y, w, **stat_kwargs)
+    attrs = []
+    all_attrs = list(dict(vars(autocorr)).keys())
+    for attribute in all_attrs:
+        attrs.append(getattr(autocorr, str(attribute)))
+    return pd.Series(attrs, index=all_attrs, name=dist)
+
+
+def correlogram(
+    gdf: gpd.GeoDataFrame,
+    variable: str,
+    distances: list,
+    distance_type: str = "band",
+    statistic: str = "I",
+    weights_kwargs: dict = None,
+    stat_kwargs: dict = None,
+    n_jobs: int = -1,
+    backend: str = "loky",
+):
+    """Generate a spatial correlogram
+
+    Parameters
+    ----------
+    gdf : gpd.GeoDataFrame
```

Comment:
While not critical for this PR (or code functionality itself), we should strive for proper spelling, grammar, and punctuation within our docstrings. English may not be the first language of many beginner users of the PySAL ecosystem so having clearly written documentation could likely help them get over the hump for using (and possibly contributing to) PySAL.

Reply:
good call